vcs/repository.go

```diff
@@ -27,8 +27,9 @@ type Repository interface {
 	// ErrBranchNotFound if no such branch exists.
 	ResolveBranch(name string) (CommitID, error)
 
-	// Branches returns a list of all branches in the repository.
-	Branches() ([]*Branch, error)
+	// Branches returns a list of all branches in the repository, as well as
+	// the total number of branches.
+	Branches(BranchesOptions) (branches []*Branch, total uint, err error)
```

Comment:
I see. That's a great point.

I think it makes sense to make `total` return value always present if and only if it's very inexpensive to return in all cases.

I can imagine that is not the case for commits. Unless the total number of commits is cached somewhere, computing it would require traversing the entire list of commits from branch tip to the root commit (one that has no more parents), one commit at a time. That's gotta be quite expensive, so it's wasteful to make the API have it as a mandatory return.

In that case, let's just remove it for now (for branches, it is currently redundant and unused information). Once we decide to add pagination support to branches, we can revisit and come up with an optimal solution. Does that sound okay?